### PR TITLE
feat: configure gas report, including skip files

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #1092 - Note uses glob instead of regex though, is that okay?

Additionally fixes bug where if two receipts had same transaction hash, the second one would not show in the gas report.

### How I did it

* Add config wrapper to handle both CLI args and `ape-config.yaml` as one and pass that as a dependency in the `pytest-plugin` code.
* Change `receipt_map` in `ReceiptCapture` to first use a key of the source ID so can trace transactions with same hash from different sources.

### How to verify it

https://github.com/fubuloubu/ERC4626/pull/18

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

**Review by fubuloubu — DISMISSED** [review]

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Would love at least one other person do QA on this feature, but looking very good!

**Comment by unparalleled-js** [comment]
tests here: https://github.com/ApeWorX/ape-hardhat/pull/108

File: src/ape_test/__init__.py [hunk 1]
```diff
@@ -9,9 +11,51 @@
 from .provider import LocalProvider
 
 
+class GasExclusion(PluginConfig):
+    contract: str = "*"  # If only given method, searches across all contracts.
+    method: Optional[str] = None
```
Comment (fubuloubu) [inline src/ape_test/__init__.py:16]: curious if these should be `contract_name` and `method_name`

Comment (fubuloubu) [inline src/ape_test/__init__.py:16]:
```suggestion
    contract: str = "*"  # If only given method, searches across all contracts.
    method: Optional[str] = None  # By default, match all methods in a contract
```

**Comment by fubuloubu** [comment]
> Would love at least one other person do QA on this feature, but looking very good!

Did some QA, works great!

**Review by fubuloubu — DISMISSED** [review]

File: src/ape/types/__init__.py [hunk 2]
```diff
@@ -64,6 +65,30 @@
...
 
 
+@dataclass
+class ContractFunctionPath:
+    """
+    Useful for identifying a method in a contract.
+    """
+
+    contract_id: str
+    method_id: Optional[str] = None
```
Comment (fubuloubu) [inline src/ape/types/__init__.py:75]: I wouldn't use `method_id` because that has a particular contextual meaning which doesn't match here

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**